kind-setup/prometheus/prometheus-setup.yaml

```diff
@@ -0,0 +1,75 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: prometheus-cluster-role
+rules:
+  - apiGroups: [""]
+    resources:
+      - nodes
+      - services
+      - endpoints
+      - pods
+    verbs: ["get", "list", "watch"]
+  - apiGroups: [""]
+    resources:
+      - configmaps
+    verbs: ["get"]
+  - nonResourceURLs: ["/metrics"]
+    verbs: ["get"]
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: prometheus-service-account
+  namespace: ack-system
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: prometheus-role-binding
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: prometheus-cluster-role
+subjects:
+  - kind: ServiceAccount
+    name: prometheus-service-account
+    namespace: ack-system
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: prometheus-deployment
+  namespace: ack-system
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: prometheus
+  template:
+    metadata:
+      labels:
+        app: prometheus
+    spec:
+      containers:
+        - name: prometheus-cont
+          image: custom-prometheus:embedded_config
+          ports:
+            - containerPort: 9090
+      serviceAccountName: prometheus-service-account
```

Comment:
Add a volume mount here instead of needing to build a special Docker image for Prometheus...

```yaml
        volumeMounts:
          - mountPath: /etc/prometheus/prometheus.yml
            name: etc-prometheus
        volumes:
          - name: etc-prometheus
            hostPath:
              path: $(PROMETHEUS_SETUP_DIR)/prometheus.yml
```

Reply:
++ We can also use a `ConfigMap` and mount it as a volume

Reply:
Updated code to use config map.